### PR TITLE
DateMiddleware

### PR DESCRIPTION
**Pull request by krisread**
This is old middleware I wrote for burgertime months ago but am copying the file here so that other projects can use it. 

It is actually ripped off from express body-parser.  No credit to me.

When inbound JSON has dates, the JSON parser does not turn them into date objects because JSON isn't typed... it just assumes its a string.   Depending on what you want to do with these dates it becomes a pain to constantly be making `new Date` every time.

This body-parer plugin matches ISO format UTC date strings using a regex (I didn't write it) and turns them into Date objects. 

This is very handy when working with a lot of dates...

### CODE REVIEW

File: src/lib/plugins/dateparser.plugin.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          return listenFn(msg);
+        }
+        return next(routingKey, parseDates, options);
+      },
+
+    publish: next =>
+      (routingKey, message, options) => next(routingKey, message, options),
+
+    subscribe: next =>
+      (routingKey, subscribeFn, options) => next(routingKey, subscribeFn, options)
```
Comment (balmasi) [inline src/lib/plugins/dateparser.plugin.js:54]: @krisread wouldn't we like to use the parser for subscribe as well?

**Comment by balmasi** [comment]
@krisread I like the concept, and as you know, I want to merge it in, but since this is bringing in moment and moment-timezone which are pretty big packages, doesn't it make sense to just publish this as a standalone plugin and not have it be an included one?

The other plugins are just a few lines of code with no dependencies but this one brings in 2 libraries

File: src/lib/plugins/dateparser.plugin.js (hunk 1 above)
Comment (krisread) [reply src/lib/plugins/dateparser.plugin.js:54]: Ah I forgot I didn't do that in burgertime because there was no subscribes

**Comment by krisread** [comment]
I guess I can see this being its own standalone plugin and also I can maybe see me doing one for Mongoose Object ID too maybe I will do that monday morning.

**Comment by balmasi** [comment]
closing to make standalone plugin

**Pull request closed without merging**